operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ControllerConfigurationOverrider.java

```diff
@@ -152,6 +156,16 @@ public ControllerConfigurationOverrider<R> withGenericFilter(GenericFilter<R> ge
     return this;
   }
 
+  public ControllerConfigurationOverrider<R> withName(String name) {
+    this.name = name;
+    return this;
+  }
+
+  public ControllerConfigurationOverrider<R> withResourceClass(Class<R> resourceClass) {
+    this.resourceClass = resourceClass;
```

Comment:
It feels weird to override a given controller configuration with a different resource so I'm not sure about this one…

Reply:
The problem is that it opens the door for improper usage so I'd rather see this support moved somewhere outside of the overrider. The goal of the overrider is to change the configuration provided by the annotations at runtime, not to create a new configuration from scratch. As such, since the resource type is mandated by the parameter type of the `Reconciler` implementation, it shouldn't be possible to override it at runtime so the more I think about it, the more against this change I am.

Reply:
Hmm, that is right. The problem is it's not possible to specify it by `@ControllerConfiguration`. Shall we put it rather there? 
Would be a better solution.

Reply:
That would be a huge change and a redundant information with the `Reconciler` implementation.

Reply:
Would be empy value. But on the other hand, users might want some hierarchy that the current approach does not detect. I guess that was the reason it was added to the starter at first place.

Reply:
I'm not sure how that would be the case… the `Reconciler` implementation should be parameterised with the resource type. Changing this just for a framework support is not a good reason to change it, especially considering that it should be a pretty rare corner case, imo.

Reply:
It's not for framework support, might be the case for anybody use just core. But we can emit this for now and see if somebody asks for it, and discuss the reasons.

Reply:
kk, let's remove this option

Reply:
revisited this again, and we would break this functionality provided by the users:
https://github.com/java-operator-sdk/operator-framework-spring-boot-starter/blob/182b1111990f99d6516d2965f54dddca8c1cf6de/starter/src/main/java/io/javaoperatorsdk/operator/springboot/starter/OperatorAutoConfiguration.java#L103-L107

if we remove this.

Reply:
Basically there is no other way to configure this through our public API, just maybe hacking it with some custom config service, what is problematic. So I think we should allow this to be configured and document it, that this is just for framework support.

Reply:
> Basically there is no other way to configure this through our public API, just maybe hacking it with some custom config service, what is problematic. So I think we should allow this to be configured and document it, that this is just for framework support.

That is not true: the current way to do it is to provide your own `ControllerConfiguration` implementation (or extend the existing `ConfigurationService` implementations) but I do agree that we could make things easier.